src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -218,6 +220,39 @@ public long FetchOffset
         // Gets or sets the offset for the next request.
         public long NextRequestOffset { get; internal set; }
 
+        public bool ConsumedOffsetOutOfRange
+        {
+            get
+            {
+                lock(consumedOffsetOutOfRangeLock)
+                {
+                    return this.consumedOffsetOutOfRangeFlag;
+                }
+            }
+            internal set
+            {
+                lock (consumedOffsetOutOfRangeLock)
+                {
+                    this.consumedOffsetOutOfRangeFlag = value;
+
+                    if (this.consumedOffsetOutOfRangeFlag)
+                    {
+                        // Clear the chunkQueue as current messages in the queue are invalid on brokers. Drop them in order to fetch valid messages in.
+                        while (this.chunkQueue.Count > 0)
```

Comment:
There is no need to empty the chunkQueue. We can leave it as it is.
-The chunkQueue is shared among many partitions that this client is dequeuing. We will lose data from other partitions.
-The partition with OutOfRange has valid data in the queue. If the consumer thread keeps up, these would get dequeued immediately anyways.

Can you explain what was your intent in dequeueing all the old (but valid) messages?